support/vm/invocation_context.go

```diff
@@ -47,6 +49,10 @@ type invocationContext struct {
 	// Used for detecting modifications to state outside of transactions.
 	stateUsedObjs map[cbor.Marshaler]cid.Cid
 	stats         *CallStats
+	// Gas trackign fields
+	gasPrices    runtime.Pricelist
```

Comment:
The fields in this struct are supposed to be specific to an inner call.

Prices don't change between inner calls, so move this to topLevelContext. I think the values below are intended to capture the whole top-level call, so move them too.